crates/holochain/src/conductor/conductor.rs

```diff
@@ -1777,9 +1777,78 @@ mod scheduler_impls {
 
 /// Miscellaneous methods
 mod misc_impls {
+    use holochain_zome_types::builder;
+    use std::collections::BTreeSet;
+
+    use crate::conductor::api::error::ConductorApiError;
+
     use super::*;
 
     impl Conductor {
+        /// Authorize a zome call signing key for a cell
+        pub async fn authorize_zome_call_signing_key(
+            &self,
+            payload: AuthorizeZomeCallSigningKeyPayload,
+        ) -> ConductorApiResult<()> {
+            let AuthorizeZomeCallSigningKeyPayload {
+                agent_pub_key,
+                cap_secret,
+                cell_id,
+                functions,
+                signing_key,
+            } = payload;
+
+            if agent_pub_key != *cell_id.agent_pubkey() {
+                return Err(ConductorApiError::IllegalZomeCallSigningKeyAuthorization(
+                    cell_id.clone(),
+                    agent_pub_key.clone(),
+                ));
+            }
+
+            let source_chain_workspace = SourceChainWorkspace::init_as_root(
+                self.get_authored_db(cell_id.dna_hash())?,
+                self.get_dht_db(cell_id.dna_hash())?,
+                self.get_dht_db_cache(cell_id.dna_hash())?,
+                self.get_cache_db(&cell_id)?,
+                self.keystore.clone(),
+                agent_pub_key.clone(),
+                Arc::new(
+                    self.get_dna_def(cell_id.dna_hash())
+                        .ok_or_else(|| ConductorApiError::DnaMissing(cell_id.dna_hash().clone()))?,
+                ),
+            )
+            .await?;
+
+            let mut assignees = BTreeSet::new();
+            assignees.insert(signing_key);
+            let cap_grant = Entry::CapGrant(ZomeCallCapGrant {
+                tag: SIGNING_KEY_TAG.to_string(),
```

Comment:
I could be wrong, but I *thought* the tag was intended to be usable by the happ ui developer, and thus should be passed through rather than hard coded.

Reply:
Actually, I'm feeling like the whole ZomeCallCapGrant should be passed in by whomever is trying to apply the grant. That would make it much more future-proof as we add additional CapAccess types, etc, the api won't need to change, and it is more generally useful.